### PR TITLE
feat(www): custom fonts in og-images + landing og dark mode

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes #

## 🎯 Changes

- adds a fetchFont that lets us use custom fonts + weights in the og image generator without keeping a bunch of large assets on the repo

- adds a og-image for the landing page in dark mode to match the rest

## ✅ Checklist

- [ ] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: www/docusaurus.config.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ module.exports = {
   themeConfig: {
     disableSwitch: false,
     respectPrefersColorScheme: true,
-    image: 'https://assets.trpc.io/www/trpc-open-graph.png',
+    image: 'https://og-image.trpc.io/api/landing',
```
Comment (juliusmarminge) [inline www/docusaurus.config.js:34]: update to env after #3402 is merged

File: www/og-image/pages/api/font.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+export const config = {
+  runtime: 'experimental-edge',
+};
+
+export default async (req: NextRequest) => {
+  if (req.nextUrl.pathname !== '/api/font') return;
+  const url = new URL(req.url);
+
+  const parsed = fontParams.decodeURL(url);
```
Comment (KATT) [inline www/og-image/pages/api/font.ts:16]: This has changed

File: www/docusaurus.config.js (hunk 1 above)
Comment (KATT) [reply www/docusaurus.config.js:37]: 👀

File: www/og-image/pages/api/font.ts (hunk 2 above)
Comment (juliusmarminge) [reply www/og-image/pages/api/font.ts]: fixed

File: www/og-image/pages/api/font.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  const res = await fetch(resource[1]);
+
+  // Make sure not to mess it around with compression when developing it locally.
+  if (url.hostname === 'localhost') {
+    res.headers.delete('content-encoding');
+    res.headers.delete('content-length');
+  }
+
+  return res;
```
Comment (KATT) [inline www/og-image/pages/api/font.ts:52]: We should probably add some cache headers to permanently cache the response

File: www/og-image/utils/zodParams.ts [hunk 4]
```diff
@@ -37,6 +37,14 @@ function truncateWords(str: string, maxWords: number) {
     : str;
 }
 
+export const fontParams = zodParams(
+  z.object({
+    family: z.string(),
+    weight: z.number().default(400),
```
Comment (KATT) [inline www/og-image/utils/zodParams.ts:43]:
```suggestion
    weight: z.enum([400, ..., ...]).default(400),
```

File: www/og-image/pages/api/font.ts (hunk 3 above)
Comment (juliusmarminge) [reply www/og-image/pages/api/font.ts:55]: good idea, done

File: www/og-image/utils/zodParams.ts (hunk 4 above)
Comment (juliusmarminge) [reply www/og-image/utils/zodParams.ts:43]: variable fonts could technically take any weight but yea we could restrict to the "default" ones

**Review by KATT — APPROVED** [review]

**Review by KATT — APPROVED** [review]

**Pull request merged**